### DISCUSSION TITLE
Manifest Tags for Linux Containers in .NET 5.0

### DISCUSSION THREAD

**Discussion by mthalman** [dotnet/dotnet-docker#3677] [Announcements]
# Manifest Tags for Linux Containers in .NET 5.0

Starting with .NET 5.0, .NET Docker images will be tagged with multi-arch manifest tags for each of the Linux distros. Prior to .NET 5.0, multi-arch tags were only available for the default Linux distro Debian. This now allows you to target a specific Linux distro's manifest tag and have it work across any of the supported architectures.

Making use of these tags can provide a better developer experience, allowing the same Dockerfile to be used across multiple machines that vary in CPU architecture. We believe this scenario will become more prevalent in the future with availability of ARM-based developer machines.

## Details

The new tagging pattern requires all simple tags for Linux distros to explicitly specify the architecture in the tag name. Prior to .NET 5.0, the absense of an architecture in a simple tag implied that the architecture was AMD64. But starting with .NET 5.0, a Linux distro tag without an architecture in the name means it is a multi-arch manifest tag.

Below are the tags that are now available with 5.0. (For brevity, the list below does not include patch version tags (e.g. `5.0.0-alpine`) but these are available to use as well. See [Docker Hub](https://hub.docker.com/_/microsoft-dotnet) for the full set.)

* `5.0-buster-slim` (multi-arch tag)
  * `5.0-buster-slim-amd64`
  * `5.0-buster-slim-arm64v8`
  * `5.0-buster-slim-arm32v7`
* `5.0-alpine` (multi-arch tag)
  * `5.0-alpine-amd64`
  * `5.0-alpine-arm64v8`
* `5.0-focal` (multi-arch tag)
  * `5.0-focal-amd64`
  * `5.0-focal-arm64v8`
  * `5.0-focal-arm32v7`

### Related Issues

* [Consider manifest tag for Alpine](https://github.com/dotnet/dotnet-docker/issues/2182)

**Comment by marijus-ravickas** [comment]
Guys, any plans to include additional tags into 3.1?

**Comment by MichaelSimons** [comment]
@marijus-ravickas, we thought about the possibility of backporting the change but we consider it a breaking change and therefore are not willing to make the change and risk breaking our customers.
